### PR TITLE
Access public history in job cache / job search

### PR DESCRIPTION
**Pull request by mvdbeek**
I think this is ready, but maybe want to make it configurable on the user side if we want to pick up public results ?

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Review by jdavcs — APPROVED** [review]
Looks fine. SQLite fix makes sense. 
Thanks!

Ready to move out of draft?

File: lib/galaxy/schema/jobs.py [hunk 1]
```diff
@@ -191,6 +191,7 @@ class EncodedJobDetails(JobSummary):
         title="Output collections",
         description="",
     )
+    user_id: Optional[EncodedDatabaseIdField] = Field(default=None, description="User ID of user that ran ran job")
```
Comment (nsoranzo) [inline lib/galaxy/schema/jobs.py]:
```suggestion
    user_id: Optional[EncodedDatabaseIdField] = Field(default=None, description="User ID of user that ran this job")
```

File: lib/galaxy_test/base/api.py [hunk 2]
```diff
@@ -147,7 +147,7 @@ def _setup_user_get_key(self, email, password=None):
         return user, self._post(f"users/{user['id']}/api_key", admin=True).json()
 
     @contextmanager
-    def _different_user(self, email=OTHER_USER, anon=False):
+    def _different_user(self, email: Optional[str] = OTHER_USER, anon=False):
```
Comment (nsoranzo) [inline lib/galaxy_test/base/api.py]:
```suggestion
    def _different_user(self, email: Optional[str] = None, anon=False):
```

File: lib/galaxy_test/base/api.py [hunk 3]
```diff
@@ -156,6 +156,7 @@ def _different_user(self, email=OTHER_USER, anon=False):
                 self._get("histories")  # Gets other_user@bx.psu.edu histories.
 
         """
+        email = OTHER_USER if email is None else email
```
Comment (nsoranzo) [inline lib/galaxy_test/base/api.py]: Move to the `else:` branch below (line 168)?

**Review by nsoranzo — APPROVED** [review]
This is great, thanks!

**Comment by bgruening** [comment]
:tada:

**Comment by jmchilton** [comment]
This seems like it really needs to be accompanied with some sort of tool syntax that declares the tool depends on the user running the job 😓.

**Comment by mvdbeek** [comment]
I haven't understood the link here, what would the tool annotation say ?

**Pull request merged**